packages/common/src/repository/activeSubmissionRepository.ts

```diff
@@ -108,14 +109,17 @@ const repository = (dependencies: Dependencies) => {
 		 * @param {any} newData Set fields to update
 		 * @returns An updated record
 		 */
-		update: async (submissionId: number, newData: Partial<Submission>): Promise<Submission> => {
+		update: async (submissionId: number, newData: Partial<Submission>): Promise<Submission | null> => {
```

Comment:
Use `undefined` instead of `null`.

I'll defend this if you want to discuss it more, but here's Microsoft's guide for developing on TypeScript itself:
https://github.com/microsoft/TypeScript/wiki/Coding-guidelines#null-and-undefined

Reply:
I've been thinking about this, and while I support using undefined over null in general context, there are valid cases for differentiating invalid values from simply missing, such as when validations failed and you cannot implement a full error reporting system. a nullified/truncated field is traditionally a good way to do so.

why am I mentioning this here? because "failure updating" is not quite the same as "function with no return", which is the default behaviour in JS. this function could do better not ever returning undefined, and just throwing in failure.

Reply:
changed to `undefined` instead of `null` in the return type.
If there's no record found to be updated in this function then it will return `undefined`. 
This will leave the obligation to the caller to handle what to do if it's an undefined value.